lib/packages.js

```diff
@@ -378,11 +378,19 @@ function _sortPackages(cache) {
   // sort packages, least dependent => most dependent (low-level to high-level)
   const unsorted = Object.keys(cache).map(name => cache[name]);
   const sorted = [];
+  const cyclicHashSet = {};
 
   while(unsorted.length > 0) {
     const pkg = unsorted.shift(1);
     const deps = Object.keys(pkg.manifest.dependencies || {});
 
+    if(cyclicHashSet[JSON.stringify(unsorted)] === true) {
```

Comment:
Do we need to stringify all of the packages or just track the package names? i.e. the `.map()` call we use in the error message below could be used to produce the key instead of stringifying the entire package contents.